daemon/cmd/state.go

```diff
@@ -82,6 +83,8 @@ func (d *Daemon) validateEndpoint(ep *endpoint.Endpoint) (valid bool, err error)
 		if err := os.RemoveAll(healthStateDir); err != nil {
 			scopedLog.Warning("Cannot clean up old health state directory")
 		}
+		// Remove health related devices if they still exist
```

Comment:
After re-reading the code, this seems like the proper place to do cleanup for now.

I think we should invoke `health.KillEndpoint()` here too. Otherwise we try to remove the network namespace of a running process, which I'm not sure works well.